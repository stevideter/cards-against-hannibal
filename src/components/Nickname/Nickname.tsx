import { useState, Dispatch, SetStateAction } from 'react';
interface NicknameProps {
    nickname: string;
    setNickname: Dispatch<SetStateAction<string>>;
}

const Nickname = (props: NicknameProps) => {
    const { nickname, setNickname } = props;
    return (
        <>
            <label>Enter Nickname:</label>
            <input
                type="text"
                value={nickname}
                onChange={(event) => setNickname(event.target.value)}
            />
        </>
    );
};

export default Nickname;
