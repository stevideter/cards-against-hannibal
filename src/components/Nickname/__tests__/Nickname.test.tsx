import Nickname from '../Nickname';
import renderer from 'react-test-renderer';

describe('Nickname', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Nickname />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
