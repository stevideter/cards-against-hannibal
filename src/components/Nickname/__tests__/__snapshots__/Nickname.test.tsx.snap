// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nickname renders correctly 1`] = `
Array [
  <label>
    Enter Nickname:
  </label>,
  <input
    onChange={[Function]}
    type="text"
    value=""
  />,
]
`;
